sdk/internal/testframework/README.md

```diff
@@ -0,0 +1,161 @@
+client secret to configure the client.
+
+```go
+//func recordedTestSetup(t *testing.T, testName string, mode testframework.RecordMode) {
+//  <...>
+    accountName, err := recording.GetRecordedVariable(storageAccountNameEnvVar, testframework.Default)
+    suffix := recording.GetOptionalRecordedVariable(storageEndpointSuffixEnvVar, DefaultStorageSuffix, testframework.Default)
+    secret, err := recording.GetRecordedVariable(storageAccountKeyEnvVar, testframework.Secret_Base64String)
+    cred, _ := NewSharedKeyCredential(accountName, secret)
+    uri := storageURI(accountName, suffix)
+```
+
+The last step is to instrument your client by replacing its transport with your `Recording` instance.
+`Recording` satisfies the `azcore.Transport` interface.
+
+```go
+//func recordedTestSetup(t *testing.T, testName string, mode testframework.RecordMode) {
+//  <...>
+    // Set our client's HTTPClient to our recording instance.
+    // Optionally, we can also configure MaxRetries to -1 to avoid the default retry behavior.
+    client, err := NewTableServiceClient(uri, cred, &TableClientOptions{HTTPClient: recording, Retry: azcore.RetryOptions{MaxRetries: -1}})
+    assert.Nil(err)
+
+    // either return your client instance, or store it somewhere that your test can use it for test execution.
+    clientsMap[testName] = &testContext{client: client, recording: recording, context: &context}
+}
+
+
+func getTestContext(key string) *testContext {
+    return clientsMap[key]
+}
+```
+
+### Completing the recorded test session
+
+After the test run completes we need to signal the `Recording` instance to save the recording.
+
+```go
+// recordedTestTeardown fetches the context from our map based on test name and calls Stop on the Recording instance.
+func recordedTestTeardown(key string) {
+    context, ok := clientsMap[key]
+    if ok && !(*context.context).IsFailed() {
+        context.recording.Stop()
+    }
+}
+```
+
+### Setting up a test to use our Recording instance
+
+Test frameworks like testify suite allow for configuration of a `BeforeTest` method to be executed before each test.
+We can use this to call our `recordedTestSetup` method
+
+Below is an example test setup which executes a single test.
+
+```go
+package aztable
+
+import (
+    "errors"
+    "fmt"
+    "net/http"
+    "testing"
+
+    "github.com/Azure/azure-sdk-for-go/sdk/internal/runtime"
+    "github.com/Azure/azure-sdk-for-go/sdk/internal/testframework"
+    "github.com/stretchr/testify/assert"
+    "github.com/stretchr/testify/suite"
+)
+
+type tableServiceClientLiveTests struct {
+    suite.Suite
+    mode         testframework.RecordMode
+}
+
+// Hookup to the testing framework
+func TestServiceClient_Storage(t *testing.T) {
+    storage := tableServiceClientLiveTests{endpointType: StorageEndpoint, mode: testframework.Playback /* change to Record to re-record tests */}
```

Comment:
`endpointType` is not part of `tableServiceClientLiveTests` struct.

```go
type tableServiceClientLiveTests struct {    
    suite.Suite    
    mode         testframework.RecordMode
}
```

Reply:
Good catch - I was trying to simplify the example, which was taken from the actual aztable package. I missed removing this reference to endpointType.